include/string_util.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <string>
+#include <algorithm>
+
+// From https://stackoverflow.com/questions/216823/how-to-trim-a-stdstring
+
+// trim from start (in place)
+static inline void ltrim(std::string &s) {
+    s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](unsigned char ch) {
+        return !std::isspace(ch);
+    }));
+}
+
+// trim from end (in place)
+static inline void rtrim(std::string &s) {
+    s.erase(std::find_if(s.rbegin(), s.rend(), [](unsigned char ch) {
+        return !std::isspace(ch);
+    }).base(), s.end());
+}
+
+// trim from both ends (in place)
+static inline void trim(std::string &s) {
+    rtrim(s);
+    ltrim(s);
+}
+
+// trim from start (copying)
+static inline std::string ltrim_copy(std::string_view s) {
```

Comment:
Since you know you're going to copy the string anyway, no need for a temporary string view or taking it by value, just take a `string` by value.

```
static inline std::string ltrim_copy(std::string s) {
    ltrim(s);
    return s;
}
```

Reply:
I just did a copy and paste of the functions.

But it is bad practice to change the input arguments of a function. (I could not find the rule in https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#main) Cppfront enforces this. So I think it should be left as it is.

Reply:
> But it is bad practice to change the input arguments of a function.

That's a very broad statement.  I don't think that's true.  That would mean that there should be no `inout` parameter passing type.

> Cppfront enforces this. 

How does it do this?  That also seems contrary to the existence of the `copy` parameter type.

>  I think it should be left as it is.

Okay.

Reply:
I agree with both @MaxSagebaum and @gregmarr 👍 

Yes, a function shouldn't modify "input"-only parameters, which are Cpp2 `in`.

Yes, if a function wants to modify its parameter, it should be either `copy` (if the parameter is effectively a local variable and modifications should be kept local) or `inout` (if the parameter is a caller-allocated variable and the modifications should be visible to the caller).